packages/dataviews/src/dataviews-layouts/grid/index.tsx

```diff
@@ -194,20 +206,24 @@ export default function ViewGrid< Item >( {
 	selection,
 	view,
 }: ViewGridProps< Item > ) {
+	const titleField = fields.find(
+		( field ) => field.id === view?.titleField
+	);
 	const mediaField = fields.find(
-		( field ) => field.id === view.layout?.mediaField
+		( field ) => field.id === view?.mediaField
 	);
-	const primaryField = fields.find(
-		( field ) => field.id === view.layout?.primaryField
+	const descriptionField = fields.find(
+		( field ) => field.id === view?.descriptionField
 	);
-	const viewFields = view.fields || fields.map( ( field ) => field.id );
-	const { visibleFields, badgeFields } = fields.reduce(
+	const otherFields = view.fields ?? [];
+	const { regularFields, badgeFields } = fields.reduce(
```

Comment:
It would only benefit from useMemo if the the component receiving this prop is memoized. I'd avoid this kind of early optimizations for now because we're not measuring that this is a perf issue but it could be something to explore once we have better dataviews performance tracking.